File: Leetcode/Climbing Stairs.cc
const int SIZE = 1024;
int dp[SIZE];

class Solution {
public:
    int climbStairs(int n) {
        memset(dp, 0, sizeof(dp));
        dp[0] = 1;
        for (int i = 1; i <= n; i++) {
            if (i - 2 >= 0) {
                dp[i] += dp[i - 2];
            }
            if (i - 1 >= 0) {
                dp[i] += dp[i - 1];
            }
        }
        return dp[n];
    }
};

File: Leetcode/python/Climbing Stairs.py
class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        stair = [0 for i in xrange(n + 1)]
        stair[0] = 1
        for i in xrange(n + 1):
            if i + 1 <= n:
                stair[i + 1] += stair[i]
            if i + 2 <= n:
                stair[i + 2] += stair[i]
        return stair[n]
